cli/src/klio_cli/commands/job/gke.py

```diff
@@ -0,0 +1,218 @@
+        image_tag = self.docker_runtime_config.image_tag
+        if image_tag:
+            dep = self.deployment_config
+            image_path = "spec.template.spec.containers.0.image"
+            # TODO: If more than one image deployed,
+            #  we need to search for correct container
+            image_base = glom.glom(dep, image_path)
+            # Strip off existing image tag if any
+            image_base = re.split(":", image_base)[0]
+            full_image = f"{image_base}:{image_tag}"
+            glom.assign(self._deployment_config, image_path, full_image)
+
+    def _apply_deployment(self):
+        """
+        Create a namespaced deploy if the deployment does not already exist.
+        If the namespaced deployment already exists then
+        `self.run_job_config.update` will determine if the
+        deployment will be updated or not.
+        """
+        dep = self.deployment_config
+        namespace = dep["metadata"]["namespace"]
+        deployment_name = dep["metadata"]["name"]
+        if not self._deployment_exists():
+            resp = self.kubernetes_client.create_namespaced_deployment(
+                body=dep, namespace=namespace
+            )
+            deployment_name = resp.metadata.name
+            logging.info(f"Deployment created for {deployment_name}")
+        else:
+            if self.run_job_config.update:
+                self._update_deployment()
+            else:
+                logging.warning(
+                    f"Cannot apply deployment for {deployment_name}."
+                    f"If deployment already exists, set `update` to True."
+                )
+
+    def _setup_docker_image(self):
+        super()._setup_docker_image()
+
+        logging.info("Pushing worker image to GCR")
+        docker_utils.push_image_to_gcr(
+            self._full_image_name,
+            self.docker_runtime_config.image_tag,
+            self._docker_client,
+        )
+
+    def run(self, *args, **kwargs):
+        # NOTE: Notice this job doesn't actually run docker locally, but we
+        # still have to build and push the image before we can run kubectl
+
+        # docker image setup
+        self._check_gcp_credentials_exist()
+        self._check_docker_setup()
+        self._setup_docker_image()
+
+        self._apply_image_to_deployment_config()
+        self._apply_deployment(**kwargs)
+
+
+class StopPipelineGKE(GKECommandMixin):
+    def __init__(self, job_dir):
+        super().__init__()
+        self.job_dir = job_dir
+
+    def stop(self):
+        """
+        Delete a namespaced deployment
+        Expects existence of a kubernetes/deployment.yaml
+        """
+        self._update_deployment(replica_count=0)
+
+
+class DeletePipelineGKE(GKECommandMixin):
+    def __init__(self, job_dir):
+        super().__init__()
+        self.job_dir = job_dir
+
+    def _delete_deployment(self):
+        dep = self.deployment_config
+        deployment_name = glom.glom(dep, "metadata.name")
```

Comment:
Some places use `glom.glom` and some places use dictionary lookups for getting info from the deployment config. Could we be consistent?